docs/scripts/reportBrokenLinks.js

```diff
@@ -0,0 +1,212 @@
+/* eslint-disable no-console */
+const path = require('path');
+const fse = require('fs-extra');
+const { createRender } = require('@mui/markdown');
+const { marked } = require('marked');
+
+const UNSUPPORTED_PATHS = ['/api/', '/careers/', '/store/', '/x/'];
+
+const docsSpaceRoot = path.join(__dirname, '../');
```

Comment:
Is there an opportunity to share the same script with the other documentation? If so, we would need to use yargs to make this path configurable or use the `cwd`.

Reply:
Since it's a single function that does the job (parsing and structuring links from a folder), I'm exporting this function and reusing it inside the X repository.

I noticed that I can access to the core documentation from X repository (thanks to `node_modules/@mui/monorepo/docs`).

So in the core repository, the script verifies internal links. And in the X repository, it checks links in both X and core.

I tested it in locale, and I'm effectively able to detect broken links between the two repository
https://github.com/mui/mui-x/pull/6497